diff -ur jalv-1.4.6/src/jalv.c jalv-1.4.6.new/src/jalv.c
--- jalv-1.4.6/src/jalv.c	2014-08-07 22:20:02.000000000 -0400
+++ jalv-1.4.6.new/src/jalv.c	2016-06-18 14:10:59.953199084 -0400
@@ -1055,9 +1055,13 @@
 	/* Create port structures (jalv.ports) */
 	jalv_create_ports(&jalv);
 
-	/* Get the plugin's name */
-	LilvNode*   name     = lilv_plugin_get_name(jalv.plugin);
-	const char* name_str = lilv_node_as_string(name);
+	/* Get the plugin's name for ports, if not specified with -n */
+	LilvNode* name = NULL;
+	const char* name_str = jalv.opts.port_name;
+	if (!name_str) {
+		name     = lilv_plugin_get_name(jalv.plugin);
+		name_str = lilv_node_as_string(name);
+	}
 
 	/* Truncate plugin name to suit JACK (if necessary) */
 	char* jack_name = NULL;
@@ -1082,7 +1086,9 @@
 	}
 
 	free(jack_name);
-	lilv_node_free(name);
+	if (name) {
+		lilv_node_free(name);
+	}
 
 	if (!jalv.jack_client)
 		die("Failed to connect to JACK.\n");
diff -ur jalv-1.4.6/src/jalv_console.c jalv-1.4.6.new/src/jalv_console.c
--- jalv-1.4.6/src/jalv_console.c	2014-08-08 01:37:10.000000000 -0400
+++ jalv-1.4.6.new/src/jalv_console.c	2016-06-18 14:12:05.852579186 -0400
@@ -35,6 +35,7 @@
 	fprintf(os, "  -h           Display this help and exit\n");
 	fprintf(os, "  -p           Print control output changes to stdout\n");
 	fprintf(os, "  -c SYM=VAL   Set control value (e.g. \"vol=1.4\")\n");
+	fprintf(os, "  -n NAME      Name for Jack ports\n");
 	fprintf(os, "  -u UUID      UUID for Jack session restoration\n");
 	fprintf(os, "  -l DIR       Load state from save directory\n");
 	fprintf(os, "  -d DIR       Dump plugin <=> UI communication\n");
@@ -72,6 +73,12 @@
 			opts->show_ui = true;
 		} else if ((*argv)[a][1] == 'p') {
 			opts->print_controls = true;
+		} else if ((*argv)[a][1] == 'n') {
+			if (++a == *argc) {
+				fprintf(stderr, "Missing argument for -n\n");
+				return 1;
+			}
+			opts->port_name = jalv_strdup((*argv)[a]);
 		} else if ((*argv)[a][1] == 'u') {
 			if (++a == *argc) {
 				fprintf(stderr, "Missing argument for -u\n");
diff -ur jalv-1.4.6/src/jalv_internal.h jalv-1.4.6.new/src/jalv_internal.h
--- jalv-1.4.6/src/jalv_internal.h	2014-08-08 01:34:51.000000000 -0400
+++ jalv-1.4.6.new/src/jalv_internal.h	2016-06-18 14:12:27.932371256 -0400
@@ -85,6 +85,7 @@
 } ControlChange;
 
 typedef struct {
+	char*    port_name;         ///< Name to use for JACK ports
 	char*    uuid;              ///< Session UUID
 	char*    load;              ///< Path for state to load
 	char**   controls;          ///< Control values
