Index: dzen.h
===================================================================
--- dzen.h	(revision 267)
+++ dzen.h	(working copy)
@@ -130,6 +130,7 @@
 	const char *bg;
 	const char *fg;
 	int line_height;
+	int text_margin;
 
 	Display *dpy;
 	int screen;
Index: main.c
===================================================================
--- main.c	(revision 267)
+++ main.c	(working copy)
@@ -256,7 +256,7 @@
 	}
 
 	if(!dzen.line_height)
-		dzen.line_height = dzen.font.height + 2;
+		dzen.line_height = dzen.font.height + dzen.text_margin * 2;
 
 	if(t->y + dzen.line_height > scr.y + scr.height)
 		t->y = scr.y + scr.height - dzen.line_height;
@@ -1023,8 +1023,11 @@
 			if(++i < argc) dzen.slave_win.width = atoi(argv[i]);
 		}
 		else if(!strncmp(argv[i], "-h", 3)) {
-			if(++i < argc) dzen.line_height= atoi(argv[i]);
+			if(++i < argc) dzen.line_height = atoi(argv[i]);
 		}
+		else if(!strncmp(argv[i], "-tm", 4)) {
+			if(++i < argc) dzen.text_margin = atoi(argv[i]);
+		}
 		else if(!strncmp(argv[i], "-tw", 4)) {
 			if(++i < argc) dzen.title_win.width = atoi(argv[i]);
 		}
