diff -rNU8 dmenu-4.1.1/dmenu.c dmenu-4.1.1.new/dmenu.c
--- dmenu-4.1.1/dmenu.c	2010-05-29 07:56:51.000000000 -0400
+++ dmenu-4.1.1.new/dmenu.c	2010-07-28 16:17:50.000000000 -0400
@@ -567,16 +567,20 @@
 		else
 			return;
 		break;
 	case XK_Tab:
 		if(!sel)
 			return;
 		strncpy(text, sel->text, sizeof text);
 		cursor = strlen(text);
+		if (cursor > 0 && cursor < sizeof text - 1) {
+			text[cursor] = ' ';
+			text[cursor + 1] = '\0';
+		}
 		match(text);
 		break;
 	}
 	drawmenu();
 }
 
 void
 match(char *pattern) {
