--- dmenu-4.0.orig/dmenu.c	2009-04-18 07:50:04.000000000 -0400
+++ dmenu-4.0/dmenu.c	2010-03-05 14:08:18.000000000 -0500
@@ -464,16 +464,21 @@ kpress(XKeyEvent * e) {
 			curr = next;
 			calcoffsets();
 		}
 		break;
 	case XK_Tab:
 		if(!sel)
 			return;
 		strncpy(text, sel->text, sizeof text);
+		num = strlen(text);
+		if (num > 0 && num < sizeof text - 1) {
+			text[num] = ' ';
+			text[num + 1] = '\0';
+		}
 		match(text);
 		break;
 	}
 	drawmenu();
 }
 
 void
 match(char *pattern) {
