--- dmenu-4.0-orig/config.h	2010-04-12 15:39:33.000000000 -0400
+++ dmune-4.0/config.h	2010-04-12 15:39:23.000000000 -0400
@@ -3,8 +3,9 @@
 /* appearance */
 static const char *font        = "-*-terminus-medium-r-normal-*-14-*-*-*-*-*-*-*";
 static const char *normbgcolor = "#cccccc";
 static const char *normfgcolor = "#000000";
 static const char *selbgcolor  = "#0066ff";
 static const char *selfgcolor  = "#ffffff";
 static unsigned int spaceitem  = 30; /* px between menu items */
 static const char *fontxft     = "Monospace-10:normal"; /*if set xft is used */  
+static unsigned int textmargin = 1;
--- dmenu-4.0-orig/dmenu.c	2010-04-12 15:39:33.000000000 -0400
+++ dmune-4.0/dmenu.c	2010-04-12 15:39:18.000000000 -0400
@@ -640,19 +640,19 @@
 
 	/* menu window */
 	wa.override_redirect = True;
 	wa.background_pixmap = ParentRelative;
 	wa.event_mask = ExposureMask | ButtonPressMask | KeyPressMask;
 
 	/* menu window geometry */
 	if(fontxft) 
-		mh = dc.xftfont.height + 2;
+		mh = dc.xftfont.height + 2 * textmargin;
 	else
-	mh = dc.font.height + 2;
+		mh = dc.font.height + 2 * textmargin;
 #if XINERAMA
 	if(XineramaIsActive(dpy) && (info = XineramaQueryScreens(dpy, &n))) {
 		i = 0;
 		if(n > 1) {
 			int di;
 			unsigned int dui;
 			Window dummy;
 			if(XQueryPointer(dpy, root, &dummy, &dummy, &x, &y, &di, &di, &dui))
@@ -751,16 +751,19 @@
 			if(++i < argc) prompt = argv[i];
 		}
 		else if(!strcmp(argv[i], "-sb")) {
 			if(++i < argc) selbgcolor = argv[i];
 		}
 		else if(!strcmp(argv[i], "-sf")) {
 			if(++i < argc) selfgcolor = argv[i];
 		}
+		else if(!strcmp(argv[i], "-tm")) {
+			if(++i < argc) textmargin = atoi(argv[i]);
+		}
 		else if(!strcmp(argv[i], "-v"))
 			eprint("dmenu-"VERSION", © 2006-2008 dmenu engineers, see LICENSE for details\n");
 		else
 			eprint("usage: dmenu [-i] [-b] [-fn <font>] [-fa <xftfont>] [-nb <color>] [-nf <color>]\n"
 			       "             [-p <prompt>] [-sb <color>] [-sf <color>] [-v]\n");
 	if(!setlocale(LC_CTYPE, "") || !XSupportsLocale())
 		fprintf(stderr, "warning: no locale support\n");
 	if(!(dpy = XOpenDisplay(NULL)))
